lib/pengine/unpack.c

```diff
@@ -560,17 +561,28 @@ unpack_nodes(xmlNode * xml_nodes, pe_working_set_t * data_set)
             new_node = NULL;
 
             id = crm_element_value(xml_obj, XML_ATTR_ID);
+            cluster_id = crm_element_value(xml_obj, PCMK_XA_CLUSTER_ID);
             uname = crm_element_value(xml_obj, XML_ATTR_UNAME);
             type = crm_element_value(xml_obj, XML_ATTR_TYPE);
             score = crm_element_value(xml_obj, XML_RULE_ATTR_SCORE);
-            crm_trace("Processing node %s/%s", uname, id);
+
+            if (cluster_id == NULL) {
+                cluster_id = id;
+            }
+
+            if (uname == NULL) {
+                uname = id;
+            }
+
+            crm_trace("Processing node %s/%s",
+                      pcmk__s(uname, "<none>"), pcmk__s(cluster_id, "<none>"));
```

Comment:
Something like "Unpacking node %s with cluster id %s and name %s" and s/"\<none\>"/"same as id"/ (before NULL checks) would be more readable